### PR TITLE
gh-113208: Mention namespace packages don't require __init__.py

### PR DESCRIPTION
**Pull request by Unique-Usman**
* Issue: gh-113208

----
📚 Documentation preview 📚: https://cpython-previews--113209.org.readthedocs.build/

### CODE REVIEW

**Comment by Unique-Usman** [comment]
@hugovk @CAM-Gerlach check this out.

File: Doc/tutorial/modules.rst [hunk 1]
```diff
@@ -437,7 +437,8 @@ When importing the package, Python searches through the directories on
 ``sys.path`` looking for the package subdirectory.
 
 The :file:`__init__.py` files are required to make Python treat directories
-containing the file as packages.  This prevents directories with a common name,
+containing the file as packages (unless using a :term:`namespace package`, a
+relatively advanced features). This prevents directories with a common name,
```
Comment (CAM-Gerlach) [inline Doc/tutorial/modules.rst]:
```suggestion
relatively advanced feature). This prevents directories with a common name,
```

Fix typo

**Review by CAM-Gerlach — COMMENTED** [review]
Thanks, @Unique-Usman . I added one suggestion to fix a typo; apply it by clicking `Commit Suggestion`, or together with others by going to the `Files changed` tab, clicking `Add suggestion to batch` for each suggestion, and then `Commit`. Otherwise, LGTM as far as my input goes

**Review by CAM-Gerlach — APPROVED** [review]
LGTM! @hugovk ?

**Comment by hugovk** [comment]
Thanks!

File: Doc/tutorial/modules.rst [hunk 2]
```diff
@@ -437,7 +437,8 @@ When importing the package, Python searches through the directories on
 ``sys.path`` looking for the package subdirectory.
 
 The :file:`__init__.py` files are required to make Python treat directories
-containing the file as packages.  This prevents directories with a common name,
+containing the file as packages (unless using a :term:`namespace package`, a
+relatively advanced feature). This prevents directories with a common name,
 such as ``string``, from unintentionally hiding valid modules that occur later
```
Comment (merwok) [inline Doc/tutorial/modules.rst:442]: So… this line is not true anymore 😢

Comment (Unique-Usman) [reply Doc/tutorial/modules.rst:442]: Hi @merwok, could you provide a reference to the updated way, I can open a pr for it.

Comment (merwok) [reply Doc/tutorial/modules.rst:442]: Well, now nothing presents a directory on sys.path to shadow a stdlib module name, even without \_\_init\_\_.py file

You could open an issue first, then we’ll see if someone creates a PR!

**Pull request merged**